compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -267,8 +270,10 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
                     ? ASTBuilderUtil.createLiteral(pos, symTable.nilType, Names.NIL_VALUE)
                     : onConflictExpr;
             BLangVariableReference tableRef = addTableConstructor(queryExpr, queryBlock);
+            BLangLiteral isReadonly = ASTBuilderUtil.createLiteral(pos, symTable.booleanType,
```

Comment:
Can move the common logic out for  `if(queryExpr.isTable) and else part`

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/36576/commits/ad38a3bc1446c773ad2bfa147be4f2cdc4be6104.